python/lsst/dm/OCPS/ocps_csc.py

```diff
@@ -130,68 +120,27 @@ class OcpsCsc(salobj.ConfigurableCsc):
 
     def __init__(
         self,
+        index,
         config_dir=None,
         initial_state=salobj.State.STANDBY,
         settings_to_apply="",
         simulation_mode=0,
     ):
         self.config = None
         self.simulated_jobs = set()
+        self.index = OcpsIndex(index)
```

Comment:
I suggest not storing the index here because it duplicates `self.salinfo.index` though admittedly will be cast to an integer (sorry; I should update ts_salobj to fix that).

Reply:
I missed looking into `salinfo` in the `Controller` attributes while searching for `index`.  Sorry.

Reply:
I'm glad it was just an oversight. The next release of salobj will save the enum "as is" instead of casting it to an int. So you _might_ want to write your code as if that was already being done, instead of explicitly turning it back into an enum value -- except you will have to print the repr instead of the `.name` property if you want to print the name of the existing index.